### PR TITLE
store/tikv: move region_* into package store/tikv/region

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR moves tikv/region_*.go into package `store/tikv/region`

Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/region.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+type StoreSelectorOption = region.StoreSelectorOption
+
+// RegionRequestRuntimeStats records the runtime stats of send region requests.
+type RegionRequestRuntimeStats = region.RequestRuntimeStats
+
+// RPCRuntimeStats indicates the RPC request count and consume time.
+type RPCRuntimeStats = region.RPCRuntimeStats
+
+// CodecPDClient wraps a PD Client to decode the encoded keys in region meta.
+type CodecPDClient = region.CodecPDClient
```
[the remark this answers is not in the document]
Comment (AndreMouche) [reply store/tikv/region.go:58]: Since we do not want the client to care about the sub-packages

File: store/tikv/region.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package tikv
```
Comment (disksing) [inline store/tikv/region.go:1]: missing license.

**Review by disksing — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**